great_expectations/rule_based_profiler/profiler.py

```diff
@@ -32,8 +32,7 @@ def __init__(
         self,
         validator: Validator,
         *,
-        variables_configs: Optional[Dict[str, Dict]] = None,
-        rules_configs: Optional[Dict[str, Dict]] = None,
+        profiler_config: Dict[str, Dict[str, Dict]] = {},
```

Comment:
I've gone ahead and made the default arg here an empty dictionary. My reasoning for this is that subsequent usages of `dict.get()` won't require unnecessary checks for `None` (since `NoneType` doesn't have a `get()` method). 

Might this have implications that I'm not considering?

Reply:
Yes! I don't know exactly what happens, but you wind up with a Class attribute (?) that will change in unexpected ways if you re-instantiate the Profiler class. (I tried to do the exact same thing pretty recently, and this was explained to me by @anthonyburdi ).

Reply:
Ah so keeping it as `None` is the recommended approach? I'll have to add checks in the method to ensure we're not using `get()` with a `NoneType` then.

@anthonyburdi would you mind telling me how that all works (or pointing me to a resource)? Would love to see what's going on behind the scenes :)

Reply:
Yup! I think we'll need to change this to None.

This provides a bit more context, I think: https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Reply:
Great link Tal!
When using mutable types as defaults for constructors in python you end up with a single instance (sort of "class" variable) that can get modified by each instance (they all use a reference to the same dict) rather than what you want which is an instance variable tied to the specific instance. It leads to tricky bugs.
Also when changing back to `None` don't forget to mark the type as `Optional`

Reply:
Really good to know, thank you both! Just when I thought I knew Python 🐍